alien/CMakeLists.txt

```diff
@@ -58,19 +58,24 @@ list(APPEND CMAKE_MODULE_PATH ${ARCFRAMEWORK_ROOT}/_common/build_all/cmake)
 
 find_package(BLAS REQUIRED)
 message(STATUS "[ Alien ] BLAS LIBRARIES: ${BLAS_LIBRARIES}")
+
 set(CBLAS_INCLUDE_DIR CBLAS_INCLUDE_DIR-NOTFOUND)
 if(BLAS_LIBRARIES MATCHES mkl OR BLA_VENDOR MATCHES Intel)
-    add_definitions(-DALIEN_USE_MKL)
     # relative path for mkl_cblas.h change with mkl versions
-    find_path(CBLAS_INCLUDE_DIR mkl_cblas.h HINTS ENV MKLROOT ENV MKL_ROOT PATH_SUFFIXES include mkl/include REQUIRED)
+    find_path(CBLAS_INCLUDE_DIR mkl_cblas.h HINTS ENV MKLROOT PATH_SUFFIXES include mkl/include REQUIRED)
+    add_definitions(-DALIEN_USE_MKL)
```

Comment:
You should use `target_add_definitions` instead because `add_definitions` is applied to ALL targets (if we compile the whole framework it may impact targets from other components)